src/ocaml/typing/typemod.ml

```diff
@@ -1931,7 +1934,7 @@ and package_constraints env loc mty constrs =
     match Mtype.scrape env mty with
     | Mty_signature sg ->
         Mty_signature (package_constraints_sig env loc sg constrs)
-    | Mty_functor _ | Mty_alias _ -> assert false
+    | Mty_functor _ | Mty_alias _ | Mty_for_hole -> assert false
```

Comment:
Can you justify why this case can't happen?

Reply:
I cannot, but wasn't able to trigger this case.
However it is probably safer to just return `Mty_for_hole` in that case.